IRremoteESP8266.cpp

```diff
@@ -260,15 +260,24 @@ void IRsend::sendWhynter(unsigned long data, int nbits) {
   space(WHYNTER_ZERO_SPACE);
 }
 
-// sendSony() should typically be called with repeat=3 as Sony devices
-// expect the code to be sent at least 3 times.
-// Timings and details are taken from:
-//   http://www.sbprojects.com/knowledge/ir/sirc.php
 void IRsend::sendSony(unsigned long data, int nbits, unsigned int repeat) {
-  // Set IR carrier frequency
-  enableIROut(40);
+  // Send an IR command to a compatible Sony device.
+  //
+  // Args:
+  //   data: IR command to be sent.
+  //   nbits: Nr. of bits of the IR command to be sent.
+  //   repeat: Nr. of additional times the IR command is to be sent.
+  //
+  // sendSony() should typically be called with repeat=2 as Sony devices
+  // expect the code to be sent at least 3 times.
+  //
+  // Timings and details are taken from:
+  //   http://www.sbprojects.com/knowledge/ir/sirc.php
+
+  enableIROut(40);  // Sony devices use a 40kHz IR carrier frequency.
   IRtimer usecs = IRtimer();
-  for (uint16_t i = 0; i < repeat; i++) {  // Typically loop 3 or more times.
+
+  for (uint16_t i = 0; i <= repeat; i++) {  // Typically loop 3 or more times.
```

Comment:
Ah, but you see, you've just proved my point, I think. :) It is precisely 'how many repeats starting 0'.

For NEC, repeat=0 means no repeats. It (NEC) sends the code, with no repeats.
For Sony, repeat=0 _needs_ to mean no repeats. It (sony) needs to send the code, with no repeats.
NEC differs because the repeat code is different to the actual command. i.e. NEC data command + N * repeat codes.
Sony just simply sends the data command multiple times for a repeat. i.e. (Sony data command) x (1 + nr. of repeats), or another way, Sony data command + (Sony data command * nr. of repeats.)